lib/faker.ex

```diff
@@ -144,29 +144,21 @@ defmodule Faker do
   end
 
   defmacro sampler(name, data) do
-    count = Enum.count(data)
-
-    mapped_data =
-      data |> Enum.with_index() |> Enum.into(%{}, fn {k, v} -> {v, k} end) |> Macro.escape()
-
     quote do
+      @count Enum.count(unquote(data))
+      @mapped_data unquote(data) |> Enum.with_index() |> Enum.into(%{}, fn {k, v} -> {v, k} end)
       def unquote(name)() do
-        unquote(mapped_data)
-        |> Map.get(Faker.random_between(0, unquote(count - 1)))
+        Map.get(@mapped_data, Faker.random_between(0, @count - 1))
```

Comment:
This is the modification mentioned in the OP.
Previously `count` and `mapped_data` calculations are applied to AST outside of `quote`, which did not allow non-literals (such as tuples) to be passed to `data`.
Pushing it inside `quote` makes them expanded at compilation of caller modules, so any expressions in `data` are properly evaluated just like other module attributes or other "outside the `def`" expressions,